src/Aspire.Dashboard/Model/InstrumentViewModel.cs

```diff
@@ -11,7 +12,7 @@ public class InstrumentViewModel
     public OtlpInstrumentSummary? Instrument { get; private set; }
     public List<DimensionScope>? MatchedDimensions { get; private set; }
 
-    public List<Func<Task>> DataUpdateSubscriptions { get; } = [];
+    public ConcurrentBag<Func<Task>> DataUpdateSubscriptions { get; } = [];
```

Comment:
It's for this error:

```
fail: Aspire.Hosting.Dashboard.Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost[111]
      Unhandled exception in circuit 'hYYAo8UMq3S58Ff2g215fTIFV_By0GPW3yIyuAYCwjo'.
      System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
         at System.Collections.Generic.List`1.Enumerator.MoveNext()
         at Aspire.Dashboard.Model.InstrumentViewModel.UpdateDataAsync(OtlpInstrumentSummary instrument, List`1 matchedDimensions) in C:\Development\Source\aspire\src\Aspire.Dashboard\Model\InstrumentViewModel.cs:line 23
         at Aspire.Dashboard.Components.ChartContainer.UpdateInstrumentDataAsync(OtlpInstrumentData instrument) in C:\Development\Source\aspire\src\Aspire.Dashboard\Components\Controls\Chart\ChartContainer.cs:line 113
         at Aspire.Dashboard.Components.ChartContainer.UpdateDataAsync() in C:\Development\Source\aspire\src\Aspire.Dashboard\Components\Controls\Chart\ChartContainer.cs:line 93
         at Aspire.Dashboard.Components.ChartContainer.DisposeAsync() in C:\Development\Source\aspire\src\Aspire.Dashboard\Components\Controls\Chart\ChartContainer.cs:line 70
         at Microsoft.AspNetCore.Components.RenderTree.Renderer.<>c__DisplayClass85_0.<<Dispose>g__HandleAsyncExceptions|1>d.MoveNext()
```

I think it is ok. An alternative could be to make the subscriptions mutable and replace it when there is a change.

Can test more later.

Reply:
Does that error only exist in metric highlights (the feature previously known as dashpages)? If so then it's fine in this PR. I thought it was happening in main, in which case the commit should be pulled out as metric highlights won't make 9.0.

Reply:
It's not happening in main.

Reply:
I haven’t ever seen it occur outside of a dashpage, that doesn’t mean it can’t still happen. James made a comment about it, which is what the fix is for